### ISSUE TITLE
Starting point of the distillation bolck

### ISSUE THREAD

**Issue by DanChen001** [Zheng222/IDN-Caffe#5]
Hi, thank you for sharing the code.

1. What is the starting point of the designed distillation block?  Why do you concate some features in the middle with the input features?  

2. It seems that the group convolution in caffe is time-consuming compared with the general convolution, right?  Although it uses less parameters and has lower computational complexity.  

3. You use many tricks in training. For example, training with small patches firstly, and fine-tuning with large patches; training with L1 loss firstly, and fine-tuning with L2 loss; using a pre-trained model to initialize the training...  

Can these tricks enhance the performance significantly? 

Thanks.

**Comment by Zheng222** [comment]
@DanChen001 Hello, for the first question, the intuition of the proposed architecture is that we notice an image not only contains smooth area (easily super-resolved) but also includes non-smooth region (difficultly super-resolved). Thus, we consider separation processing of the feature maps and divide preceding feature maps into two parts, where the small part represents the processing completed in this phase and the rest indicates the more difficult part that requires to further process. The output feature maps of the third convolution in each enhancement unit are visualized as follows:

![](https://github.com/Zheng222/IDN-Caffe/raw/master/files/visualization.jpg)

Taking the last subfigure as an example, we easily see that the feature maps in red box are mostly smooth and simple.

Answer to Q2: The group convolution is slightly slow, but you can try other deep learning development kit (e.g. Pytorch, Tensorflow).

Answer to Q3: One point to note is that the pre-trained model is trained with 291 dataset, we don't have to add extra training dataset. These tips can improve a little performance.
